fcm-channel/src/main/java/io/fcmchannel/sdk/util/AttachmentHelper.java

```diff
@@ -0,0 +1,44 @@
+package io.fcmchannel.sdk.util;
+
+import androidx.core.util.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class AttachmentHelper {
+
+    private static final String REGEX_IMAGE = "(http|https)://.*\\.(jpg|jpeg|png|gif|tif|tiff|bmp)";
+    private static final String REGEX_VIDEO = "(http|https)://.*\\.(mp4|mkv|wmv|avi|flv)";
+
+    public static String getUrlFileExtension(final String url) {
+        final int index = url.lastIndexOf(".") + 1;
+        return index <= url.length() - 1 ? url.substring(index) : "";
+    }
+
+    public static boolean isImageUrl(final String url) {
```

Comment:
Where is the case when the image comes at the end of the text message, especially in push notifications?

Reply:
The handling is in `BundleHelper` and the `extractMediaUrls` method from this class is used. That method returns a pair with first element being the text without urls and second with the extracted urls.

Reply:
Perfect